carapace-server/src/main/java/org/carapaceproxy/server/mapper/requestmatcher/OrRequestMatcher.java

```diff
@@ -0,0 +1,54 @@
+/*
+ Licensed to Diennea S.r.l. under one
+ or more contributor license agreements. See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership. Diennea S.r.l. licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+
+ */
+package org.carapaceproxy.server.mapper.requestmatcher;
+
+import io.netty.handler.codec.http.HttpRequest;
+import java.util.List;
+import org.carapaceproxy.server.config.RoutingKey;
+
+/**
+ *
+ * @author paolo.venturi
+ */
+public class OrRequestMatcher implements RequestMatcher {
+
+    private final List<RequestMatcher> matchers;
+
+    public OrRequestMatcher(List<RequestMatcher> matchers) {
+        this.matchers = matchers;
+    }
+
+    @Override
+    public RoutingKey matches(HttpRequest request) {        
+        for (RequestMatcher matcher: matchers) {
+            RoutingKey result = matcher.matches(request);
+            if (result != null) {
+                return result;
```

Comment:
we have to evaluate all of the childer and aggregate the 'attributes' of the matching ones